Sources/ServiceLifecycle/ServiceGroupConfiguration.swift

```diff
@@ -107,7 +107,7 @@ public struct ServiceGroupConfiguration: Sendable {
     public var cancellationSignals = [UnixSignal]()
 
     /// The group's logger.
-    public var logger: Logger
+    public var logger: Logger?
```

Comment:
We can't make this change since this an API break. I think at this point there is no nice way to make this optional anymore sadly.

Reply:
I see, I've reverted the changes of making it an optional, with only improved docs and readme examples remaining 👍

Reply:
An alternative approach to make the logger in the initializers optional could be to chain the log handler from the passed in optional logger to the stored-property logger:
```swift
self.logger = Logger(logger.label ?? "ServiceLifecycle", factory: { _ in logger?.handler ?? SwiftLogNoOpLogHandler() })
```